src/github.com/getlantern/flashlight/app/settings.go

```diff
@@ -145,6 +145,14 @@ func (s *Settings) checkBool(data map[string]interface{}, name string, f func(bo
 	}
 }
 
+func (s *Settings) checkInt(data map[string]interface{}, name string, f func(int)) {
+	if v, ok := data[name].(int); ok {
```

Comment:
Do these tests consider that this is a JSON message unmarshalled into a `map[string]interface{}`? https://github.com/getlantern/lantern/blob/214e2822948dac3a36f56714a72fb01055d231bf/src/github.com/getlantern/flashlight/app/settings.go#L122

My point is that when we unmarshal a JSON Number into an interface, it becomes a float64: https://play.golang.org/p/NiHuolVKvo

Hence the type check and the subsequent cast to int.